script.js

```diff
@@ -1,4 +1,238 @@
+var dicePosition = [];
+var diceValues = [];
+
+// Helper function to list the indexes in the array
+var getDicePosition = function (diceLength) {
+  var index = 0;
+  while (index < diceLength) {
+    dicePosition.push(index);
+    index = index + 1;
+  }
+  return dicePosition;
+};
+
+// Helper function to list the random dice roll values
+var getNumOfDiceAndRoll = function (diceLength) {
+  var index = 0;
+  while (index < diceLength) {
+    diceValues.push(randomDiceRoll());
+    index = index + 1;
+  } return diceValues;
+};
+
+// Helper function to generate concatenated number after player's index choice
+var generatePlayerScore = function (input) {
+  var positions = input.split('');
+  var playerNumberArray = [];
+  var index = 0;
+  while (index < diceValues.length) {
+    playerNumberArray.push(diceValues[positions[index]]);
+    index = index + 1;
+  }
+  playerScore = playerNumberArray.join('');
+  return playerScore;
+};
+
+// Start Game
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = '';
+  var invalidMessage = 'Please enter a valid position number';
+  // User to choose which mode to play: Normal or Variable Dice Mode
+  if (gameMode == 'Enter Mode') {
+    if (input == 'normal') {
+      myOutputValue = 'You have chosen the normal mode. Click submit to roll dice.';
+      gameMode = 'normal';
+    }
+    else if (input == 'variable') {
+      myOutputValue = 'Please enter the qty of dice you want.';
+      gameMode = 'Player 1 Variable Dice Mode';
+    }
+    else { myOutputValue = 'Please enter "normal" or "variable" to select mode.';
+    }
+    return myOutputValue;
+  }
+
+  // Ask player1 how many dice they want to play with
+  if (gameMode == 'Player 1 Variable Dice Mode') {
+    if (isNaN(Number(input)) == false && input != '') {
+      numOfDice = input;
+      // List the index position of the dice values in order using the helper function
+      // Match it to Player 1
+      dicePosition = getDicePosition(numOfDice);
+      p1DicePosition = dicePosition;
+      // List the dice values using the helper function
+      // Match it to Player 1
+      diceValues = getNumOfDiceAndRoll(numOfDice);
+      p1DiceValues = diceValues;
+      gameMode = 'Player 1 Enter Positions';
+      // Output Player 1 dice values and prompt index question.
+      myOutputValue = 'Player 1, you have decided to <br>play with ' + numOfDice + ' dice. <br><br>You rolled: <br>---- dice: ' + p1DiceValues + '<br>position: ' + p1DicePosition + '<br><br>Enter the positions of the numbers you want.';
+    } else {
+      myOutputValue = invalidMessage;
+    }
+    return myOutputValue;
+  }
+
+  if (gameMode == 'Player 1 Enter Positions') {
+  // Check if input is a number and is not an empty string.
+    if (isNaN(Number(input)) == false && input != '') {
+      // Put input into helper function to generate player's score
+      generatePlayerScore(input);
+      player1Score = Number(playerScore);
```

Comment:
`playerScore` does not need to be a global. It can be a local variable returned by `generatePlayerScore`, e.g. `player1Score = generatePlayerScore(input)`